configtype/duration.go

```diff
@@ -2,29 +2,161 @@ package configtype
 
 import (
 	"encoding/json"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
 	"time"
 
 	"github.com/invopop/jsonschema"
 )
 
+var (
```

Comment:
I prefer if we don't change the semantics of the existing duration type, as that could breaking existing behavior for consumers of this type

Reply:
All previously value values are still value for the duration type. This just adds more possible valid values. Additionally all previously valid values will serialize to the same string after this PR in its current state. (unless I missed something)

Reply:
> This just adds more possible valid values.

Exactly, so now the JSON validation is more relaxed, so something that would have failed before can pass now.
While this could be seen as a feature, it's better if plugins opt-in to it, then document the new behavior